File: 04.Simulation模拟/leetcode_3761_minMirrorPairDistance.cpp
#include <iostream>
#include <vector>
#include <string>
#include <climits>
#include <unordered_map>
using namespace std;

class Solution {
private:
    int reverseNum(int x){
        string s = to_string(x);
        int y = 0;
        for (int i=s.size()-1; i>=0; i--){
            y = y*10+(s[i]-'0');
        }
        return y;
    }
public:
    int minMirrorPairDistance(vector<int>& nums) {
        int n = nums.size();
        int res = INT_MAX;
        unordered_map<int, int> pos;
        for (int i=0; i<n; i++){
            if (pos.count(nums[i])){
                res = min(res, i-pos[nums[i]]);
            }
            pos[reverseNum(nums[i])] = i;
        }
        return res==INT_MAX ? -1 : res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {12,21,45,33,54};
    cout << s.minMirrorPairDistance(nums) << endl;
}

File: 04.Simulation模拟/leetcode_3761_minMirrorPairDistance.py
from typing import List
from collections import defaultdict

class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        n = len(nums)
        res = 10**18
        pos = defaultdict(int)
        for i in range(n):
            if nums[i] in pos:
                res = min(res, i-pos[nums[i]])
            pos[int(str(nums[i])[::-1])] = i
        return -1 if res==10**18 else res

if __name__ == "__main__":
    s = Solution()
    nums = [12,21,45,33,54]
    print(s.minMirrorPairDistance(nums))
